build_ut.sh

```diff
@@ -20,37 +20,80 @@
 # *
 
 # Clone the Unit Testing Suit for this repo, it follows a standard convention
-# HAL Name is replaced with haltest, allowing this script to be part of the template for all
+# halif Name is by default to halif-test, allowing this script to be part of the template for all hals
+# This is a general configuration for ease, you can hard code it if that's the requirement
 
 # This will look up the last tag in the git repo, depending on the project this may require modification
-TEST_REPO=$(git remote -vv | head -n1 | awk -F ' ' '{print $2}' | sed 's/halif/halif-test/g')
+# We need to check for the old format, and also support either old or new depending on the repo.
+# Transition was from hal -> hal-test, new format is from halif -> halif-test
+# More Info : https://github.com/rdkcentral/ut-core/wiki/2.3.-Standards:-Performing-Changes-to-an-interface
 
-# Set default UT_PROJECT_VERSION to master
+# This script aims to provide the latest binary compatible version of the testing suite for a clean environment.
+# It supports automatic upgrades within the same major version (e.g., 1.x.x to 1.y.y). However, when a new major
+# version is released (e.g., 2.x.x), manual intervention is required to update UT_PROJECT_VERSION.
+# This precaution prevents potential test breakages caused by incompatible changes introduced in the new major version.
+# The script will choose highest version of the tags starting with ${UT_PROJECT_VERSION}. If it fails to find one, then
+# it falls back to the highest version available among tags
+
+TEST_REPO=$(git remote -vv | head -n1 | awk -F ' ' '{print $2}' | sed 's/halif/halif-test/g' | sed 's/hal-/haltest-/g' )
+DIR="."
+
+# Setting default UT_PROJECT_VERSION to 1.
 if [ -z "${UT_PROJECT_VERSION}" ]; then
-    UT_PROJECT_VERSION=master
+    UT_PROJECT_VERSION=main #alternatively use : UT_PROJECT_VERSION=main ./build_ut.sh, to force build version main/master
```

Comment:
this should never be main.

it should be set currently to the version of the latest active tag on main, which is currently `1.3.`, but every time you perform a release, this value must be MODIFIED.

can you add a note

/* NOTE: Every time a release is perform on the HALIF, e.g. 1.5.0, you must change this value to the same or greater than the release so "1.5." so that when you checkout the 1.5.0 the tag, it will pull the correct 1.5.x testing suite which is aligned with the interface".. If you do NOT Bump this version, then checking out the tag of the HALIF, you will not get a testing suite that covers all the functions */